### ISSUE TITLE
compatibile with THE FRAME 2020 QE55LS03TAUXZT

### ISSUE THREAD

**Issue by pilot1981** [ollo69/ha-samsungtv-smart#45]
I would integrate HA with THE FRAME 2020 QE55LS03TAUXZT

I would start and stop TV, change channel and volume...

I installed this integration via HACS, but I'm not able to integrate it...

this is already compatibile with THE FRAME 2020 QE55LS03TAUXZT?

can you help me?

**Comment by Riggaa** [comment]
I just added my 2020 47" frame to HA with this integration. 

1) Make sure you generate a smartthings token and add it during the set up sequence when you add the Host IP. (If you're not in front of the tv this my be difficult, helps to have someone else watch the TV while you do this step. [Token info link](https://www.home-assistant.io/integrations/smartthings/)
2) Make sure to accept the prompt on the TV to allow home assistant to control the TV right after you.

I don't believe this integration has the ability to switch to art mode though.

**Comment by ollo69** [comment]
> I don't believe this integration has the ability to switch to art mode though.

This is implemented, power button should be able to switch between modes, but I don't have a frame model so I cannon guarantee that everything work properly. Refer to issue #19 for detail on what was already done.

**Comment by skynet01** [comment]
it shows it as a status attribute, so you can check if art mode is running or not but i don't think you can turn it on into art mode (this would be amazing btw)

**Comment by pilot1981** [comment]
@Riggaa you are able to switch between art mode and power on mode?

**Comment by Riggaa** [comment]
> @Riggaa you are able to switch between art mode and power on mode?

No, I do not have the ability to switch to art mode in HA, just normal Samsung TV control.

**Comment by pilot1981** [comment]
so, this custom components isn't able to manage art mode of FRAME models?

this custom component works fine with homekit integration of HA?

**Comment by pilot1981** [comment]
Hi,
I installed it via HCAS on my HASSIO 2020.12.1 on Rpi3b+, integration via UI don't works, so I configured it via configuration.yaml:

samsungtv_smart:
  - host: 192.168.4.151
    api_key: "98529d37-xxxxxxxxxxx-7fbf073eb0a6"
    device_id: "1362c781-xxxxxxxxxxx-3931eb8366b0"
    device_name: SamsungTV
    mac: "a4:30:7a:e7:7a:f0"
    app_list: '{"Netflix": "11101200001", "YouTube": "111299001912"}'
    show_channel_number: True
    broadcast_address: 192.168.4.255

at HASSIO reboot, I authorized it directly on TV, but then I don't find any new media_player entity...which name it could have?

**Comment by ollo69** [comment]
> > @Riggaa you are able to switch between art mode and power on mode?
> 
> No, I do not have the ability to switch to art mode in HA, just normal Samsung TV control.

The component is able to manage art_mode, but the power command is designed to always turn on or off the device. This means that:

- if tv is in art_mode, the switch command turn it off
- if tv is in normal_mode, the switch command turn it off
- if tv is off, the switch command turn it on

I don't have a frame TV, so please explain what is the current behavior and how you would expect it work, I can review the switch command sequence and control.

**Comment by ollo69** [comment]
> Hi,
> I installed it via HCAS on my HASSIO 2020.12.1 on Rpi3b+, integration via UI don't works, so I configured it via configuration.yaml:

Do not use configuration.yaml for initial configuration, it only create problem. Remove any row in condiguration.yaml, reboot and configure using UI. If it doesn't work check the HA log for error.
While you configure the TV be sure that TV is on and remove any authorization, it should ask for authorization on TV screen during setup. If you want to use smartthings, be sure that your TV is configured in phone app.
If your TV is on a different subnet (HA installed in docker) the component doesn't work, this is an issue related to the WS library.

**Comment by skynet01** [comment]
@ollo69 is there anything I can do to help you add the art control functionality? Maybe capture some packets? I think art mode is just like any app on that tv. Maybe it can be switched on just like Netflix app?

**Comment by ollo69** [comment]
> @ollo69 is there anything I can do to help you add the art control functionality? Maybe capture some packets? I think art mode is just like any app on that tv. Maybe it can be switched on just like Netflix app?

As I know is not as any app, it have a dedicated control channel. But as I explained, I'm managing it. The question is what you are expecting? There is an attribute that should provide the art mode status. Does it work?

**Comment by skynet01** [comment]
The attribute status works great! Thank you for that!  I wanted to have automation when I enter the room tv is turned on and goes to art mode automatically, currently, that's not possible since I can only turn it on or off remotely.

**Comment by ollo69** [comment]
So you need a command to set the TV in art_mode right? May be this could be a service?
How you set the art_mode from TV remote when TV is off? There is a dedicated button or you press power button multiple time?

**Comment by skynet01** [comment]
Service would be awesome! 
- Power button press > art mode
- Power button press again > fully on
- Power button press again > back to art mode
- Power button hold > fully off

Even using the Samsung Smart Things app creates the same behavior above. Here is event details (from Smarthing Website) when I turn on the tv using Samsung Smart Things app and it goes to art mode:
![image](https://user-images.githubusercontent.com/8117544/103158558-412e4400-477c-11eb-9a8a-8d0769288236.png)

**Comment by ollo69** [comment]
I think that you can switch fully on -> art mode (and vice versa) simply using service `media_player.play_media` with this paramter:

```
  "entity_id": "media_player.<your entity>",
  "media_content_type": "send_key",
  "media_content_id": "KEY_POWER"
```

Turn on in art mode is not possible because turn on is manage by WOL, there is no control on this. But you can check the artstatus mode attribute and than send the command explained before.

**Comment by skynet01** [comment]
Hmm doesn't seem to work get "Unsupported media type" in the log when I send it. I tried using the service tab and converted it to YAML syntax:
`
entity_id: media_player.office_tv
media_content_id: 'send_key'
media_content_type: "KEY_POWER"
`

**Comment by ollo69** [comment]
You inverted `media_content_id` and `media_content_type`! And do not quote `KEY_POWER`

```
entity_id: media_player.office_tv
media_content_type: 'send_key'
media_content_id: KEY_POWER
```

**Comment by skynet01** [comment]
...oh jeez, thank you for noticing. That worked, thank you :) You can add it to the official documentation on how to turn on art mode :)

**Comment by pilot1981** [comment]
> So you need a command to set the TV in art_mode right? May be this could be a service?
> How you set the art_mode from TV remote when TV is off? There is a dedicated button or you press power button multiple time?

for me it can be useful switch from art mode to on mode...

to troubleshoot initial configuration using web UI, how I can setup debug logs?

**Comment by ollo69** [comment]
> for me it can be useful switch from art mode to on mode...

Call service `media_player.turn_on`

> to troubleshoot initial configuration using web UI, how I can setup debug logs?

Add the following line in `configuration.yaml`:

```
logger:
  default: info
  logs:
    custom_components.samsungtv_smart: debug
```

**Comment by pilot1981** [comment]
sorry, where I have to put it: media_player.turn_on?


In lovelace, I have this one:
type: media-control
entity: media_player.samsung_tv

In homekit integration, I have only media_player.samsung_tv

**Comment by ollo69** [comment]
`media_player.turn_on` is a service. You can call it from lovelace using a custom control like [mini media player](https://github.com/kalkih/mini-media-player) that allow you to create additional button and associate services.

https://github.com/kalkih/mini-media-player

**Comment by pilot1981** [comment]
if some time media_player.turn_on service don't works, what suggest to do? see logs?

**Comment by ollo69** [comment]
Yes, enable debug and check logs. There are some specific condition when service don't work?

**Comment by pilot1981** [comment]
I'm not sure...I'm looking g for logs...

**Comment by skynet01** [comment]
@pilot1981 are you getting any errors? To turn on the tv by default you should be able to just press power button on the entity that's created. To turn on just the art mode which is what this threat is about you need to call a play_media service. 

Here is how to do this:
Go to developer tools > services (homeassistant-url/developer-tools/service) in the service drop-down select: 
media_player.play_media
in the entities drop down select your Frame TV. Should be something like
_media_player.tv_name_

then in the service data field add 2 additional lines as per ollo69's comment so the data field should be:
```
entity_id: media_player.tv_name
media_content_type: 'send_key'
media_content_id: KEY_POWER
```
Press call service, If your tv is in the off mode sending this will turn on art mode. If you press it again tv should turn on. If this works you can now set up automation or scripts that follow the same logic. So you can go to the automation screen and create a trigger (ex: motion detection) followed by a service call (repeat steps above)

**Comment by PaleoW** [comment]
Hi,

I don't know if this is due to a different model, version or configuration, but in my case, when I press the Power/Art key on my remote when the Frame TV is completely off, the last state the TV was on is restored.

For example, if I turned off the TV while it was on HDMI1, if I press Power, or send the KEY_POWER keys through play_media, the TV turns back ON on HDMI1. Pushing KEY_POWER again then turns it to Art mode.

If the TV was left in Art mode then powered off, or was manually powered off, then Art mode will start after pushing Power.

The way you guys are talking , it seems like in your case , whatever state the TV was on when turned off doesn't matter, and Art mode ALWAYS start when pushing Power. Is that correct ? 

I went through my TV settings and didn't see anything that could explain this. 

This makes managing Art mode impossible with HA. I was looking at turning ON Art Mode in the morning, but if the TV was left blasting on the Set-top box the night before, the automation will turn on the TV and wake everyone up. I've tried sending the KEY_POWER twice when the art_mode_status attribute = off, but the sounds comes up really quick on this TV so it's too slow anyway.

Basically at the moment, there are no confirmed method to send a proper signal to the TV to get into Art mode, whatever state it's in at the moment ? 

L

**Comment by mindtripper** [comment]
> Hi,
> 
> I don't know if this is due to a different model, version or configuration, but in my case, when I press the Power/Art key on my remote when the Frame TV is completely off, the last state the TV was on is restored.
> 
> For example, if I turned off the TV while it was on HDMI1, if I press Power, or send the KEY_POWER keys through play_media, the TV turns back ON on HDMI1. Pushing KEY_POWER again then turns it to Art mode.
> 
> If the TV was left in Art mode then powered off, or was manually powered off, then Art mode will start after pushing Power.
> 
> The way you guys are talking , it seems like in your case , whatever state the TV was on when turned off doesn't matter, and Art mode ALWAYS start when pushing Power. Is that correct ?
> 
> I went through my TV settings and didn't see anything that could explain this.
> 
> This makes managing Art mode impossible with HA. I was looking at turning ON Art Mode in the morning, but if the TV was left blasting on the Set-top box the night before, the automation will turn on the TV and wake everyone up. I've tried sending the KEY_POWER twice when the art_mode_status attribute = off, but the sounds comes up really quick on this TV so it's too slow anyway.
> 
> Basically at the moment, there are no confirmed method to send a proper signal to the TV to get into Art mode, whatever state it's in at the moment ?

Hi @PaleoW 

I have the QE65LS03TAUXXC (2020 model) and my TV behaves like yours.

Using the following switch template, I can set the TV to Art Mode. No matter what the previous state was.

```
switch:
  - platform: template
    switches:
      tv_art_mode:
        friendly_name: "TV Art Mode"
        value_template: "{{ state_attr('media_player.samsung_the_frame_65_tizen', 'art_mode_status') == 'on' }}"
        turn_on:
          - service: media_player.turn_on
            entity_id: media_player.samsung_the_frame_65_tizen
          - wait_template: "{{ state_attr('media_player.samsung_the_frame_65_tizen', 'art_mode_status') == 'off' or state_attr('media_player.samsung_the_frame_65_tizen', 'media_content_type') == 'video' or state_attr('media_player.samsung_the_frame_65_tizen', 'app_id') == 'TV/HDMI' }}"
            timeout: '00:02:00'
            continue_on_timeout: false
          - delay: '00:00:01'
          - service: media_player.play_media
            data_template:
              entity_id: media_player.samsung_the_frame_65_tizen
              media_content_id: 'KEY_POWER'
              media_content_type: 'send_key'
        turn_off:
          - service: media_player.turn_off
            entity_id: media_player.samsung_the_frame_65_tizen
```

**Comment by ollo69** [comment]
I just release a new version that implement a dedicated service `set_art_mode`. Please try it and let me know if it works.

**Comment by skynet01** [comment]
works for me, thank you for doing this @ollo69 it really makes a difference :)

**Comment by mindtripper** [comment]
Works on my 2020 65".
Loving this component more and more. Thank you!

**Comment by pilot1981** [comment]
my tv change always ip because  go from wired to wireless connection and from wireless to wired connection. so I have trouble to use this module...any suggestion?

I already tr to use only wired cable, but something go wrong on the tv and smarthing connection....there is a workaround?

**Comment by Letrab** [comment]
> my tv change always ip because go from wired to wireless connection and from wireless to wired connection. so I have trouble to use this module...any suggestion?
> 
> I already tr to use only wired cable, but something go wrong on the tv and smarthing connection....there is a workaround?

Map a static IP address to your smart TV! Either in your router(/ISP modem) configuration (map MAC address to IP in DHCP settings).

Or try to set a static IP address in the Network Settings of the TV (don't know if this is possible).

**Comment by pilot1981** [comment]
@Letrab this isn't possible to assign same ip address to both nic because they have different mac address...you have same mac address for both?

**Comment by Letrab** [comment]
Don't know really. You could check your DHCP reservation list? But for me it seems the mac-address is the same for both.
Either way, I checked: you can set a static IP on your TV as well, so you could try that.
Or try to fix the wired connection with Smartthings? Works perfectly here, so no need to switch constantly. There are additional settings in the Network setup menu for allowing to control your TV, so check that out.

But this is quite off-topic for this issue. Hope you'll figure it out.

**Comment by ollo69** [comment]
> my tv change always ip because go from wired to wireless connection and from wireless to wired connection. so I have trouble to use this module...any suggestion?
> 
> I already tr to use only wired cable, but something go wrong on the tv and smarthing connection....there is a workaround?

I do not understand how is possible that your TV change automatically from wired to wireless, as I know you must specify in TV configuration which kind of connection you wand to use. May be your problem is that router is assigning different IP to your TV when you turn it on? In this case you must set a static IP in the TV configuration or create a DHCP reservation in your router.

**Comment by pilot1981** [comment]
Hi,
I'm having some issue with homekit integration of Samsung TV; I go to explain better:

1) I started from scratch homekit integration with TV and other items:
2) I obtained in integration section of HA 2 homekit integration cards: 1 for samsunt TV and 1 for HA bridge
3) I rebooted HA
4) I see only HA bridge (I don't see anymore card for integration of only Samsung TV)
5) I joined HA bridge in apple home app, but I don't have samsung TV in my homekit installation


what you suggest to see Samsung TV in my homekit installation?


regards

**Comment by pilot1981** [comment]
for now I fixed problem on homekit integration, integrating only samsung tv (no any other HA entity).

Now, on my Samsung TV FRAME I have 2 mac address with 2 IP address: there is a way to config single TV with 2 mac address and 2 ip address?

**Comment by pilot1981** [comment]
> I just release a new version that implement a dedicated service `set_art_mode`. Please try it and let me know if it works.

@ollo69 please, can you say howto use service 'set_art_mode'?

**Comment by ollo69** [comment]
> please, can you say howto use service 'set_art_mode'?

What do you mean? Is a HA service, just call it with the TV entity and the TV should be set in art mode, as the service name suggest.

**Comment by pilot1981** [comment]
If I have service samsungtv_smart.set_art_mode on entity media_player.samsung_tv how I can create a switch to it on lovelace or in Homekit?

**Comment by ollo69** [comment]
This is not the right place (BTW this is a closed thread).
Anyway a simply solution is to use [mini media player](https://github.com/kalkih/mini-media-player), for other solution refer to HA documentation and forum
